File: Templates/File Templates/Rambler Templates/Viper Module.xctemplate/___FILEBASENAME___ModuleAssembly.m
//
//  ___FILENAME___
//  Project:   ___PROJECTNAME___
//
//  Module:   ___VARIABLE_viperModuleName___
//  Description: ___VARIABLE_viperModuleDescription___
//
//  By ___FULLUSERNAME___  ___DATE___
//  ___ORGANIZATIONNAME___ ___YEAR___
//

#import "___VARIABLE_viperModuleName___ModuleAssembly.h"
#import "___VARIABLE_viperModuleName___ViewController.h"
#import "___VARIABLE_viperModuleName___Interactor.h"
#import "___VARIABLE_viperModuleName___Presenter.h"
#import "___VARIABLE_viperModuleName___Router.h"

@implementation  ___VARIABLE_viperModuleName___ModuleAssembly

- (___VARIABLE_viperModuleName:identifier___ViewController *)view___VARIABLE_viperModuleName:identifier___ {

    return [TyphoonDefinition withClass:[___VARIABLE_viperModuleName:identifier___ViewController class]
                          configuration:^(TyphoonDefinition *definition) {
                            [definition injectProperty:@selector(output) 
                                                  with:[self presenter___VARIABLE_viperModuleName:identifier___]];
             }];
}

- (___VARIABLE_viperModuleName:identifier___Interactor *)interactor___VARIABLE_viperModuleName:identifier___ {

    return [TyphoonDefinition withClass:[___VARIABLE_viperModuleName:identifier___Interactor class]
                          configuration:^(TyphoonDefinition *definition) {
                            [definition injectProperty:@selector(output) 
                                                  with:[self presenter___VARIABLE_viperModuleName:identifier___]];
             }];
}

- (___VARIABLE_viperModuleName:identifier___Presenter *)presenter___VARIABLE_viperModuleName:identifier___ {

    return [TyphoonDefinition withClass:[___VARIABLE_viperModuleName:identifier___Presenter class]
                          configuration:^(TyphoonDefinition *definition) {
                            [definition injectProperty:@selector(view) 
                                                  with:[self view___VARIABLE_viperModuleName:identifier___]];                                            
                            [definition injectProperty:@selector(interactor) 
                                                  with:[self interactor___VARIABLE_viperModuleName:identifier___]];
                            [definition injectProperty:@selector(router) 
                                                  with:[self router___VARIABLE_viperModuleName:identifier___]];
            }];
}

- (___VARIABLE_viperModuleName:identifier___Router *)router___VARIABLE_viperModuleName:identifier___ {
    return [TyphoonDefinition withClass:[___VARIABLE_viperModuleName___Router class]];
}

@end

File: Templates/File Templates/Rambler Templates/Viper Module Swift.xctemplate/___FILEBASENAME___ModuleAssembly.swift
//
//  ___FILENAME___
//  Проект:   ___PROJECTNAME___
//
//  Модуль:   ___VARIABLE_viperModuleName___
//  Описание: ___VARIABLE_viperModuleDescription___
//
//  Создан ___FULLUSERNAME___  ___DATE___
//  ___ORGANIZATIONNAME___ ___YEAR___
//

import UIKit
import Typhoon

public class ___FILEBASENAMEASIDENTIFIER___ModuleAssembly: TyphoonAssembly {

	enum ViperSEL: Selector {
        case view = "view"
        case interactor = "interactor"
        case router = "router"
        case output = "output"
    }

	public dynamic func view___FILEBASENAMEASIDENTIFIER___() -> AnyObject {
		return TyphoonDefinition.withClass(___FILEBASENAMEASIDENTIFIER___View.self) {
			(definition) in 
			
			/*
			//use for init from nib
			definition.useInitializer("initWithNibName:bundle:") {
				(initializer) in

				initializer.injectParameterWith(TyphoonConfig("//XibName//"))
				initializer.injectParameterWith(NSBundle.mainBundle())
			}*/

			definition.injectProperty(ViperSEL.output.rawValue, with: self.presenter___FILEBASENAMEASIDENTIFIER___())
		}
	}


	public dynamic func interactor___FILEBASENAMEASIDENTIFIER___() -> AnyObject {
		return TyphoonDefinition.withClass(___FILEBASENAMEASIDENTIFIER___Interactor.self){
            (definition) in
            
            definition.injectProperty(ViperSEL.output.rawValue, with: self.presenter___FILEBASENAMEASIDENTIFIER___())
        }
	}


	public dynamic func presenter___FILEBASENAMEASIDENTIFIER___() -> AnyObject {
		return TyphoonDefinition.withClass(___FILEBASENAMEASIDENTIFIER___Presenter.self){
            (definition) in
            
            definition.injectProperty(ViperSEL.view.rawValue, with: self.view___FILEBASENAMEASIDENTIFIER___())
            definition.injectProperty(ViperSEL.interactor.rawValue, with: self.interactor___FILEBASENAMEASIDENTIFIER___())
            definition.injectProperty(ViperSEL.router.rawValue, with: self.router___FILEBASENAMEASIDENTIFIER___())
        }
	}


	public dynamic func router___FILEBASENAMEASIDENTIFIER___() -> AnyObject {
		return TyphoonDefinition.withClass(___FILEBASENAMEASIDENTIFIER___Router.self){
            (definition) in
           
        }
	}
}